exercises/practice/rational-numbers/rational-numbers.spec.wren

```diff
@@ -0,0 +1,225 @@
+import "wren-testie/testie" for Testie, Expect
+import "./rational-numbers" for Rational
+
+class Math {
+  static approxEqual(a, b) { (a - b).abs <= 1e-6 }
```

Comment:
I feel like `expectToApproxEqual` (or something) might be slightly better... thoughts?

Of course this (something like it) really belongs in `testie`, but that's a whole chore. :-)

Reply:
Hmm, as it returns a bool, it's a predicate function. "expect" implies (to me) that it has some test framework responsibilities. Particularly since I put it in a Math class.

Playing around with it. How would you feel about:
```wren
class MoreExpect is Expect {
  static value(v) { MoreExpect.new(v) }
  construct new(v) { _value = v }
  toBeApproxEqual(v) { toBeApproxEqual(v, 1e-6) }
  toBeApproxEqual(v, epsilon) { (_value - v).abs <= epsilon }
}
```
and then
```
    do.test("Raise a real number to a zero rational number") {
      var r = Rational.new(0, 1)
      MoreExpect.value(r.expreal(2)).toBeApproxEqual(1.0)
    }
```

It's at times like this I miss Ruby's refinements (or even monkeypatching)

Reply:
I think when I mentioned this I WAS imagining that you'd wrap `Expect`, so it would no longer return a boolean... a whole 5 line class is not terrible but....

> It's at times like this I miss Ruby's refinements 

At first I was asking myself if Wren had dynamic dispatch, but I think no... although you could register custom expectations with a map maybe?

```js
Expect.value(r).to["beApproxEqual"](3.14)
// or would you need a call as well?
Expect.value(r).to["beApproxEqual"].call(3.14)
```

I forget... and it's a bit blah either way. :)

Reply:
This isn't anything that *has* to change... I was just being nitpicky. :)

Reply:
OK, one more crack:
```
class ExpectExt is Expect {
  static toBeApproxEqual(a, b) {
    Expect.value((a - b).abs <= 1e-6).toBe(true)
  }
}
```
```
    do.test("Raise a real number to a zero rational number") {
      var r = Rational.new(0, 1)
      ExpectExt.toBeApproxEqual(r.expreal(2), 1.0)
    }
```

or something a bit more fun and readable
```
class ExpectExt is Expect {
  static toBeApproxEqual(expected, expr) {
    var actual = expr.call()
    Expect.value((expected - actual).abs <= 1e-6).toBe(true)
  }
}
```
```
    do.test("Raise a real number to a zero rational number") {
      var r = Rational.new(0, 1)
      ExpectExt.toBeApproxEqual(1.0) { r.expreal(2) }
    }
```

Reply:
I think the block form is a little weird... though mostly I don't love the `ExpectExt` name. :-)  We can just go with what you had originally.

Reply:
In that case, we're ready to merge.